src/mc-rangeopts.c

```diff
@@ -324,6 +351,141 @@ bool mc_RangeOpts_appendMax(const mc_RangeOpts_t *ro,
     return true;
 }
 
+// Used to calculate max trim factor. Returns the number of bits required to represent any number in
+// the domain.
+bool mc_getNumberOfBits(const mc_RangeOpts_t *ro,
+                        bson_type_t valueType,
+                        uint32_t *bitsOut,
+                        mongocrypt_status_t *status) {
+    BSON_ASSERT_PARAM(ro);
+    BSON_ASSERT_PARAM(bitsOut);
+
+    // For each type, we use getTypeInfo to get the total number of values in the domain (-1)
+    // which tells us how many bits are needed to represent the whole domain.
+    // note - can't use a switch statement because of -Werror=switch-enum
+    if (valueType == BSON_TYPE_INT32) {
+        int32_t value = 0;
+        mc_optional_int32_t rmin = {false, 0}, rmax = {false, 0};
+        if (ro->min.set) {
+            value = bson_iter_int32(&ro->min.value);
+            rmin = OPT_I32(value);
+            rmax = OPT_I32(bson_iter_int32(&ro->max.value));
```

Comment:
```suggestion
            BSON_ASSERT (ro->max.set);
            rmax = OPT_I32(bson_iter_int32(&ro->max.value));
```

Suggest asserting `ro->max` is set before access. If `min` is set and `max` is not set, I expect an error will be returned before in `mc_RangeOpts_to_FLE2RangeInsertSpec` and `mc_RangeOpts_parse`.

Same suggestion applies to other accesses of `ro->max` in this function.

Reply:
Fixed.